zorg/buildbot/builders/UnifiedTreeBuilder.py

```diff
@@ -204,8 +204,15 @@ def addNinjaSteps(
         step_name = "build-{}unified-tree".format(step_name)
         step_description.extend(["unified", "tree"])
 
+    # Helper to for use truncating step names to 50 chars, needed due to
+    # <https://github.com/buildbot/buildbot/issues/3414>.
+    def trunc50(name):
+        if len(name) > 50:
+            return name[:47] + "..."
```

Comment:
I was going to suggest `f"{s:.47s}..."` but this adds `...` regardless of length, so don't do that.

Just mentioning that in case someone else was thinking of that.